pkg/skaffold/build/cluster/secret.go

```diff
@@ -53,6 +51,14 @@ func (b *Builder) setupPullSecret(out io.Writer) (func(), error) {
 		return func() {}, nil
 	}
 
+	if _, err := secrets.Get(b.PullSecretName, metav1.GetOptions{}); err == nil {
```

Comment:
Can we remove code duplication and add a method `CreateOrRecreateSecrets(secrentName string)`

It will also be easier to test this

Reply:
Ya sure

Reply:
hi Tejal,

IMHO its better to keep them seperate for readability. We can coalesce both the function but somehow I feel its much cleaner.